archive/miu_language/decision_nec.lean

```diff
@@ -0,0 +1,246 @@
+/-
+Copyright (c) 2020 Gihan Marasingha. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Gihan Marasingha
+-/
+import .basic
+import data.nat.modeq
+import tactic.ring
+
+/-!
+# Decision procedure: necessary condition
+
+We introduce a condition `decstr` and show that if a string `en` is `derivable`, then `decstr en`
+holds.
+
+Using this, we give a negative answer to the question: is `"MU"` derivable?
+
+## Tags
+
+miu, decision procedure
+-/
+
+namespace miu
+
+open miu_atom nat list
+
+/-!
+### Numerical condition on the `I` count
+
+Suppose `st : miustr`. Then `count I st` is the number of `I`s in `st. We'll show, if
```

Comment:
```suggestion
Suppose `st : miustr`. Then `count I st` is the number of `I`s in `st`. We'll show, if
```

(I found this using my [doc previewer](https://observablehq.com/@bryangingechen/github-lean-doc-preview?url=%223739%22#docs).)